### PR TITLE
Initialize app and add tests

### PR DESCRIPTION
**Pull request by greganswer**
### Proposed Changes
- Initialize a Ruby on Rails API project
- Add Member and Friendship Request Specs so that candidate submissions can be reviewed with less manual verification of app functionality
- Add **Reviewer Setup** section to the README for reviewer to execute
- Add binstubs to simplify setup

### For Reviewer
- You can review each commit individually if preferred

### CODE REVIEW

File: spec/requests/members_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    end
+
+    it 'returns an array' do
+      subject
+      expect(body).to be_an_instance_of(Array)
+    end
+  end
+
+  describe 'viewing a member' do
+    # context 'when member exists' do
```
Comment (katie-wright) [inline spec/requests/members_spec.rb]: Should this be un-commented?

File: README.md [hunk 2]
```diff
@@ -49,3 +49,21 @@ We encourage the use of any libraries for everything except the search functiona
...
+## Reviewer Setup
+
+1. Clone the candidate's branch
+1. Execute the following:
+
+        bin/rails db:drop
+        bin/setup
+        bin/rspec
+        
+1. [Make a copy](https://docs.google.com/spreadsheets/u/0/d/1t6aY1dpVv4jEG1Xq2x5DHkQrXNsMrJfn_fhdSZM6C_8/copy) of the rubric and fill it in
```
Comment (katie-wright) [inline README.md:74]: This won't be accessible by those outside the organization, right?

File: README.md [hunk 3]
```diff
@@ -49,3 +49,21 @@ We encourage the use of any libraries for everything except the search functiona
 * Remember to make meaningful commits as you work
 * Somehow share your repository with us
 * __Important:__ If there are credentials required (.env or master.key file), please email these to us directly or we can’t review your project
+* After you fork the repo, you can run the following commands in your terminal:
+        
+        bin/setup
+        bin/rspec
+        
+* Feel free to add to add/modify existing specs. Just make sure the original request specs are still present.
```
Comment (katie-wright) [inline README.md:57]: Should we add a note in the instructions that the routes and params should be named as per the existing request specs? I've seen a ton of variation in naming so I wouldn't want someone to start on a solution without realizing these have already been set.

File: spec/requests/members_spec.rb (hunk 1 above)
Comment (greganswer) [reply spec/requests/members_spec.rb]: I was debating how to handle this section without guiding the candidate too much (such as model factories). What do you think of something like this:

```ruby
    context 'when member exists' do
      subject { get "/members/#{Member.first.id}", headers: headers }

      it 'returns the correct status code' do
        subject
        expect(response).to have_http_status(:success)
      end
    end
```
If the above doesn't work, then I can just remove this test completely. We'll still gain some value from having the other tests.

File: README.md (hunk 2 above)
Comment (greganswer) [reply README.md:74]: Correct. I tested it with my personal account. It says I need to request access.

File: README.md (hunk 3 above)
Comment (greganswer) [reply README.md:57]: I was thinking if they read the instructions and run `bin/rspec` they would see something like the following:
```
ActionController::RoutingError:
       No route matches [GET] "/members"
```
If they don't read the instructions, that might be a couple points against the candidate. On the other hand, I remember the stress of trying to do a take home code challenge, among other things. I can add the routes to the README.

File: spec/requests/members_spec.rb (hunk 1 above)
Comment (katie-wright) [reply spec/requests/members_spec.rb]: Yeah I think that spec works! Super simple but proves that they implemented the route, and they could add more to it as they work on the feature

File: README.md (hunk 3 above)
Comment (katie-wright) [reply README.md:57]: Agreed I think listing the routes is a small nudge to get them moving in the right direction and will probably make our lives easier!

**Review by katie-wright — APPROVED** [review]
I am excited about this, I think having the rails app initialized and some hints for the candidate in the request specs will help with getting started and hopefully lead to higher quality submissions (while also making it easier for the reviewer) 🏆

**Review by balthatrix — APPROVED** [review]
LGTM! 

One thing that we may consider depending on how much pain others have felt on this is using SQLite instead of Postgres if we're going to template that part into the challenge.

I've had a couple instances where initializing the app db screwed my DB server up locally and it took me an extra 30 minutes to track that down and fix.

**Comment by greganswer** [comment]
> I've had a couple instances where initializing the app db screwed my DB server up locally and it took me an extra 30 minutes to track that down and fix.

@balthatrix I added the following to the Reviewer section of the README

        bin/rails db:drop
        bin/setup

This should work consistently with every reviewer who has a postgresql setup locally. If you know the exact issue you had or if others feel strongly about it, we can move this sqlite. I just have to re-test the tests (with actual code). I'm flexible, though I would prefer postgresql

**Comment by Ricar7o** [comment]
Saw that this was merged already, but wanted to comment on it, since we received an email from Dependabot about vulnerabilities.
image
Prob. worth addressing, while we're improving the repo.

**Pull request merged**